### PR TITLE
DHFPROD-2592: tab delimited support for ingest csv

### PR DESCRIPTION
**Pull request by paxtonhare**
DHFPROD-2592
also fixes DHFPROD-2635

### CODE REVIEW

**Comment by ayuwono** [comment]
> On QuickStart:
> 
> * There is no label
> * I can only put one character on the input field. The separator is \t
> 
> Screen Shot 2019-06-12 at 2 51 29 PM

But it seems that the separator is keystroke "tab"

**Comment by aebadirad** [comment]
This should resolve the issues found previous and make it more user friendly at the same time

**Comment by wooldridge** [comment]
Made a few more updates to the separator setting with my latest commits. Also performed manual tests with all four separator types, ingestions all worked: [sep-types.zip](https://github.com/marklogic/marklogic-data-hub/files/3283671/sep-types.zip)

File: marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java [hunk 1]
```diff
@@ -280,6 +283,13 @@ public RunStepResponse run() {
         inputFilePath = (String)fileLocation.get("inputFilePath");
         inputFileType = (String)fileLocation.get("inputFileType");
         outputURIReplacement = (String)fileLocation.get("outputURIReplacement");
+        if(inputFileType.equalsIgnoreCase("csv") && fileLocation.get(separator) != null) {
```
Comment (srinathgit) [inline marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java]: The "if" condition has to be 
if(inputFileType.equalsIgnoreCase("csv") && fileLocation.get("separator") != null)

Comment (hao1st) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java]: https://github.com/marklogic/marklogic-data-hub/blob/574d4b7f2ac4c9dd0bb6f9fb2ccb6bda3d10c0b8/marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java#L286-L291

 line 291 separator cannot be null in the if block. We can check like:
csvColumnSeparator = separator.isNotEmpty ? separator.charAt(0) :  ',';

Comment (hao1st) [reply marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java]:
```suggestion
we do not need a member variable "csvColumnSeparator", instead
private char csvColumnSeparator = ','; => private static final CSV_DEFAULT_COLUMN_SEPARATOR = ','; 
       separator = Character.toString(',');
        if (inputFileType.equalsIgnoreCase("csv") && fileLocation.get("separator") != null) {
           String tmp =((String) fileLocation.get("separator")).trim(); 
           if (tmp.equals("\\t") {
               separator = "\t";
           }
        }
        
     https://github.com/marklogic/marklogic-data-hub/blob/574d4b7f2ac4c9dd0bb6f9fb2ccb6bda3d10c0b8/marklogic-data-hub/src/main/java/com/marklogic/hub/step/impl/WriteStepRunner.java#L627
     
     changed line 627 => .withColumnSeparator(separator.charAt(0));
```

**Pull request merged**